app/controllers/admin/users_controller.rb

```diff
@@ -119,37 +119,26 @@ def penalty_history
   end
 
   def suspend
-    SuspendUser.call(user: @user) do
+    User::Suspend.call do
       on_success do
         render_json_dump(
           suspension: {
             suspend_reason: result.reason,
-            full_suspend_reason: result.user_history&.details,
-            suspended_till: @user.suspended_till,
-            suspended_at: @user.suspended_at,
+            full_suspend_reason: result.full_reason,
+            suspended_till: result.user.suspended_till,
+            suspended_at: result.user.suspended_at,
```

Comment:
Here, we rely on the model contained in the result object instead of relying on the one fetched by the controller.